resolvers/node/resolution.rs

```diff
@@ -61,48 +61,66 @@ use crate::NpmPackageFolderResolver;
 use crate::PackageJsonResolverRc;
 use crate::PathClean;
 
-pub static DEFAULT_CONDITIONS: &[&str] = &["deno", "node", "import"];
-pub static REQUIRE_CONDITIONS: &[&str] = &["require", "node"];
-static TYPES_ONLY_CONDITIONS: &[&str] = &["types"];
+pub static DEFAULT_CONDITIONS: &[Cow<'static, str>] = &[
+  Cow::Borrowed("deno"),
+  Cow::Borrowed("node"),
+  Cow::Borrowed("import"),
+];
+pub static REQUIRE_CONDITIONS: &[Cow<'static, str>] =
+  &[Cow::Borrowed("require"), Cow::Borrowed("node")];
+static TYPES_ONLY_CONDITIONS: &[Cow<'static, str>] = &[Cow::Borrowed("types")];
 
 #[allow(clippy::disallowed_types)]
 type ConditionsFromResolutionModeFn = crate::sync::MaybeArc<
-  dyn Fn(ResolutionMode) -> &'static [&'static str] + Send + Sync + 'static,
+  dyn Fn(ResolutionMode) -> &'static [Cow<'static, str>] + Send + Sync,
 >;
 
-#[derive(Default, Clone)]
-pub struct ConditionsFromResolutionMode(Option<ConditionsFromResolutionModeFn>);
+#[derive(Clone)]
+pub enum ConditionsResolver {
+  FromResolutionModeFn(ConditionsFromResolutionModeFn),
+  List(Vec<Cow<'static, str>>),
```

Comment:
I feel like this still needs to take into account the resolution mode? We should probably still just have one variant here but make it work off non-`'static` references.

![image](https://github.com/user-attachments/assets/010eb57c-ccb9-42f7-aaeb-26cbe94e9d73)

Reply:
Or actually it might be good to have the resolve function here handle this